### PR TITLE
Fix iOS simulator build

### LINKED ISSUE #9999: OpenSSL 1.1.1d broke iOS simulator build
Hello,

bisecting showed that commit 87bea6550ae0dda7c40937cff2e86cc2b0b09491 broke the iOS simulator build (between 1.1.1c and 1.1.1d).

Running the following on `OpenSSL_1_1_1-stable`:

```
./Configure debug-iossimulator-xcrun
make -j$(nproc)

...

Undefined symbols for architecture x86_64:
  "_aesni_cbc_encrypt", referenced from:
      _aesni_init_key in e_aes.o
      _aesni_cbc_cipher in e_aes.o
```

Tested on macOS 10.14 with Xcode 11.

Thanks,
Gregor

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Fixes #9999

### CODE REVIEW

File: crypto/evp/e_aes.c [hunk 1]
```diff
@@ -176,7 +176,7 @@ static void ctr64_inc(unsigned char *counter)
 # define HWAES_xts_decrypt aes_p8_xts_decrypt
 #endif
 
-#if     !defined(OPENSSL_NO_ASM) &&                     (  \
+#if     defined(OPENSSL_CPUID_OBJ) &&                   (  \
```
Comment (slontis) [inline crypto/evp/e_aes.c]: OPENSSL_CPUID_OBJ seems to come from configdata.pm (for libcrypto and provider.fips).
Does that mean it is always set?

Comment (bernd-edlinger) [reply crypto/evp/e_aes.c]: Yes, the rationale is the code below uses OPENSSL_ia32cap_P, therefore the CPUID_OBJ need to be
included.  This fixes build targets without assembler support like the ios, or linux-generic32
which has the same issue, unfortuately I always used linux-generic32 together with no-asm so
I did not notice earlier.

Comment (slontis) [reply crypto/evp/e_aes.c]: Master is ok by the looks of it..

**Comment by bernd-edlinger** [comment]
Merged to 1.1.1 as 6f3ccda.  Thanks!

**Pull request closed without merging**